plugins/tradeMetricsHandler.go

```diff
@@ -16,27 +16,19 @@ func MakeTradeMetricsHandler() *TradeMetricsHandler {
 	}
 }
 
-// Reset sets the handler's trade counter to zero.
+// Reset sets the handler's trades to empty.
 func (h *TradeMetricsHandler) Reset() {
 	h.trades = []model.Trade{}
 }
 
-// Read stores new trades internally.
-func (h *TradeMetricsHandler) Read(newTrades []model.Trade) {
-	for _, nt := range newTrades {
-		h.trades = append(h.trades, nt)
-	}
-}
-
-// NumTrades returns the number of trades.
-func (h *TradeMetricsHandler) NumTrades() int {
-	return len(h.trades)
+// GetTrades returns all stored trades.
+func (h *TradeMetricsHandler) GetTrades() []model.Trade {
+	return h.trades
 }
 
-// TotalBaseVolume returns the total base volume.
-func (h *TradeMetricsHandler) TotalBaseVolume() (total float64) {
-	for _, t := range h.trades {
-		total += t.Volume.AsFloat()
-	}
-	return
+// HandleFill handles a new trade
+// Implements FillHandler interface
+func (h *TradeMetricsHandler) HandleFill(trade model.Trade) error {
```

Comment:
we need to test this tradeMetricsHandler. For now let's have one simple test case, where we add 3 trades:
- buy XLM/USDT
- buy XLM/USDT
- sell XLM/USDT

then we assert that the computed values on the struct returned from `ComputeTradeMetrics` is correct.

this `TestComputeTradeMetrics` would be very simple and would ensure our calculations are correct.